### PR TITLE
PR: SK VectorDB connector work - Merging Forked branch & PR from Fork to SK branch

### PR DESCRIPTION
**Pull request by tawalke**
### Motivation and Context

To provide SK connector for Qdrant Vector database using memory architecture of kernel 

This pull request/PR review is to add the ability for the Semantic Kernel to persist embeddings/memory to external vector databases like Qdrant. This submission has modifications and additions to allow for integration into the current SK Memory architecture and the subsequent SDK/API of various Vector Databases. The VectorDB Skill/Connectors has significant changes which are likely more than initial estimations.

Please help reviewers and future users, providing the following information:

1. Why is this change required?
This change is required in order to allow SK developers/users to persist and search for embeddings/memory from a Vector Database like Qdrant. 

2. What problem does it solve?
Adds capabilities of Long Term Memory/Embedding storage to a vector databases from the Semantic Kernel. 

3. What scenario does it contribute to?
Scenario: Long term, scalability memory storage, retrieval, search, and filtering of Embeddings. 

4. If it fixes an open issue, please link to the issue here.
N/A

### Description

This PR currently includes connection for the Qdrant VectorDB only. What is out of scope: This PR removes the initial Milvus VectorDB addition and generic VectorDB client interfaces used for consistency across various external vector databases. This concept will be provided in forthcoming design & PR. 

**Addition and Modification of custom SK Qdrant.Dotnet SDK**
* Removal of VectorRecord and VectorMetaData replacing with VectorRecordData which inheirits from IEmbeddingwithMetadata, IDataStore, IEmbeddingIndex
* Adding FetchVectorsRequest and FetchVectorsResponse classes for calls the Points/Scroll API of Qdrant vector database to get ALL vectors *without* vectorid by collection name only. Initially not in original custom SDK
* Update of QdrantDB constructor for connection to add port so Qdrant API for both REST/gRPC (which have same calls per Qdrant) are supported for performance and binary data
* Adding Points internal class to support various Qdrant Points API calls
* Adding a default vectorsize if not passed in that is default as ADA model.
* Update of IVectorDbCollection.cs to support VectorRecordData class and DataEntry<VectorRecordData>
* Adding method to QdrantCollections for GetAllVectorsAsync method
* Update to QdrantCollections.cs to support updated IVectorDBCollections interface changes.
* Changing almost add signatures of Qdrant Methods to return DataEntry<VectorRecordData<float>> instead of VectorRecord
* Changing of SearchVectorsResponse.cs
* Adding FetchAllCollectionNamesRequest and FetchAllCollectionNamesResponse classes for calls the List Collection API for Qdrant Vector database to get vectors *without* vectorid by collection name only. Initial not in SDK

**Additons to Skills.Memory.VectorDB**
* Adding new namespace: Skills.Memory.VectorDB
* Adding Qdrant VectorDB client for SK, QdrantVectorDB 
* Adding ILongTermMemoryStore interface for VectorDB
* Creating/Adding Qdrant Memory Store class, QdrantMemoryStore.cs. Adding new method for connecting, retrieving collections and embedding from Qdrant DB in cloud. 

These notes will help understanding how your code works. Thanks!

***Note***
* This does build but it has several warnings as the existing SDK code that was unchanged has separate Logger and other functionality that SK now provides that I am requesting review of possible removal.  - Based upon comments in fork, several of these have been addressed.
* Question about the need or method for GetCollections to retrieve ALL collections in a external database which could be significant data request as external vector databases store embeddings:  Would like to discuss possible established limit from SK on pull.  Adding API from Qdrant REST API to handle Getting Collection names. 

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [X ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by tawalke** [comment]
Doing Diff check to ensure all updates/committed files were successfully merged from Fork.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Diagnostics/Exception.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Qdrant.DotNet.Internal.Diagnostics;
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Diagnostics/Exception.cs:5]: Lots of namespace issues. This will need to be standardized everywhere... I'm not going to call out each one, as they've already been pointed out in previous PRs.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/VectorRecord.cshold [hunk 2]
```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/VectorRecord.cshold:2]: VectorRecord.cshold?

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/_Qdrant.Dotnet.csproj_ [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/_Qdrant.Dotnet.csproj_:2]: file name \_Qdrant.Dotnet.csproj_?

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Storage;
+using Qdrant.DotNet;
+using Qdrant.DotNet.Internal;
+using Qdrant.DotNet.Internal.Diagnostics;
+
+
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs:16]: dotnet format

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        this._qdrantclient = new QdrantVectorDb(host, port);
+    }
+
+    public async IAsyncEnumerable<string> GetCollectionsAsync(CancellationToken cancel = default)
+    {
+        yield return null;
+    }
+
+
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs:42]: docs needed on all public methods of public classes

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.hold [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.hold:2]: QdrantMemoryStore.hold

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Skills.Memory.VectorDB.csproj [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    <PropertyGroup>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+        <RepoRoot>$([System.IO.Path]::GetDirectoryName($([MSBuild]::GetPathOfFileAbove('.gitignore', '$(MSBuildThisFileDirectory)'))))</RepoRoot>
+    </PropertyGroup>
+    <Import Project="$(RepoRoot)/dotnet/nuget/nuget-package.props" />
+
+    <PropertyGroup>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+        <AssemblyName>Microsoft.SemanticKernel.Skills.Memory.VectorDB</AssemblyName>
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Skills.Memory.VectorDB.csproj:11]: Same feedback from previous PR:
Microsoft.SemanticKernel.Skills.Memory.Qdrant

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs [hunk 8]
```diff
@@ -64,19 +63,6 @@ public class VolatileMemoryStore<TEmbedding> : VolatileDataStore<IEmbeddingWithM
...
-    /// <returns>A tuple consisting of the <see cref="IEmbeddingWithMetadata{TEmbedding}"/> cosine similarity result.</returns>
-    private (IEmbeddingWithMetadata<TEmbedding>, double) PairEmbeddingWithSimilarity(Embedding<TEmbedding> embedding,
-        IEmbeddingWithMetadata<TEmbedding> embeddingWithData)
-    {
-        var similarity = embedding.Vector.ToArray().CosineSimilarity(embeddingWithData.Embedding.Vector.ToArray());
-        return (embeddingWithData, similarity);
-    }
-
     #endregion
 }
```
Comment (shawncal) [inline dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs:67]: Microsoft.SemanticKernel.Skills.Memory.VectorDB

**Review by shawncal — CHANGES_REQUESTED** [review]

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Diagnostics/Exception.cs (hunk 1 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Diagnostics/Exception.cs:5]: @shawncal these were address in other files per other PR copying these files back over to ensure this PR is updated correctly.

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs [hunk 9]
```diff
@@ -3,7 +3,6 @@
 using System.Collections.Generic;
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs:3]: Hey @tawalke please confirm these deletions are expected.

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs:1]: These files and other in this test project are bad merge results, I believe you just need to delete this and BackendConfigTests, OpenAIConfigTests, and CodeBlockTests.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Skills.Memory.Qdrant.csproj [hunk 11]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <RepoRoot>$([System.IO.Path]::GetDirectoryName($([MSBuild]::GetPathOfFileAbove('.gitignore', '$(MSBuildThisFileDirectory)'))))</RepoRoot>
+    </PropertyGroup>
+    <Import Project="$(RepoRoot)/dotnet/nuget/nuget-package.props" />
+
+    <PropertyGroup>
+        <AssemblyName>Microsoft.SemanticKernel.Skills.Memory.VectorDB</AssemblyName>
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Skills.Memory.Qdrant.csproj:8]: As Shawn noted, update this namespace.

File: dotnet/src/IntegrationTest/AI/OpenAIEmbeddingTests.cs [hunk 12]
```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
```
Comment (lemillermicrosoft) [inline dotnet/src/IntegrationTest/AI/OpenAIEmbeddingTests.cs:2]: Bad merge result, revert these changes.

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs (hunk 10 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs:1]: Okay will do.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+/// <summary>
+/// An implementation of <see cref="ILongTermMemoryStore{TEmbedding}"/> which is an embedding of type float 
+/// backed by a Qdrant Vector database. The type of Embedding (float) to be stored in this data store which has the interface of 
+/// <see cref="IDataStore{VectorRecordData}"/>
+/// </summary>
+/// <remarks>The Embedding data is saved to a Qdrant Vector database instance specified in the constructor by url and port.
+/// The embedding data persists between subsequent instances and has similarity search capability.
+/// </remarks>
+public class QdrantMemoryStore : ILongTermMemoryStore<float>
```
[the remark this answers is not in the document]
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs]: @shawncal ILongTermMemoryStore inherits IMemoryStore; This was added in collaboration with @awharrison-28 when working on Memory working with collaboration with Qdrant.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Qdrant/QdrantMemoryStore.cs [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+public class QdrantMemoryStore : ILongTermMemoryStore<float>
+{
+    public QdrantMemoryStore(string host, int port)
+    {
+        this._qdrantClient = new QdrantVectorDb(host, port);
+    }
+
+    public IAsyncEnumerable<string> GetCollectionsAsync(CancellationToken cancel = default)
+    {
+        return AsyncEnumerable.Empty<string>();
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Qdrant/QdrantMemoryStore.cs:34]: Implement this.

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs (hunk 9 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs]: There should have been no changes to VolatileMemoryStore.cs. This must be a bad merge issue.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Skills.Memory.Qdrant.csproj (hunk 11 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Skills.Memory.Qdrant.csproj:8]: Okay copying local version which has this change into commit.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantDataStore.hold [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+
+namespace Microsoft.SemanticKernel.Skills.Memory.VectorDB;
+
+/// <summary>
+/// /// An implementation of <see cref="IDataStore{TValue}"/> backed by a Qdrant Vector database.
+/// </summary>
+/// <remarks>The vector data is saved to Qdrant Vector Database run locally or in cloud, specified in the constructor.
+/// The data persists between subsequent instances. </remarks>
+/// <typeparam name="TValue">The type of data to be stored in this data store.</typeparam>
+public class QdrantDataStore<TValue> : IDataStore<TValue>
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantDataStore.hold]: Qdrant will only ever hold vectors. There should be no QdrantDataStore, only QDrantMemoryStore

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/ILongTermMemoryStore.cs [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Storage;
+using Qdrant.DotNet;
+
+namespace Microsoft.SemanticKernel.Skills.Memory.VectorDB;
+
+internal interface ILongTermMemoryStore<TEmbedding> : IQdrantMemoryStore<TEmbedding>, IMemoryStore<TEmbedding>
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/ILongTermMemoryStore.cs]: Unnecessary interface. QDrantMemoryStore should implement IMemoryStore

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantVectorDb.cs [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+
+
+namespace Microsoft.SemanticKernel.Skills.Memory.VectorDB;
+
+/// <summary>
+/// An implementation of a client for the Qdrant VectorDB. This class is used to 
+/// connect, create, delete, and get embeddings data from a Qdrant VectorDB instance.
+/// </summary>
+
+public class QdrantVectorDb
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantVectorDb.cs]: I would like to see this renamed to QdrantVectorDBClient to indicate that this wraps the Qdrant API

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs (hunk 10 above)
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs]: Tests are under the SemanticKernel.UnitTest project now. Addressing the project scoping should reduce some added files in this PR

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs [hunk 18]
```diff
@@ -3,7 +3,6 @@
...
 using System.Linq;
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs]: Not sure what's going on here. I think a fork sync needs to be done. VolatileMemoryStore should not be changed for this PR

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Http/Specs/CreateCollectionRequest.cs [hunk 19]
```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Http/Specs/CreateCollectionRequest.cs]: Suggestion: rename namespace to HttpSchema to match the conventions in this repository

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs [hunk 20]
```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs]: Please ensure that QdrantMemoryStore() can seamlessly replace VolatileMemoryStore() in all of our samples. Example: Example15_MemorySkill.cs

**Review by awharrison-28 — CHANGES_REQUESTED** [review]

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs (hunk 8 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs]: @shawncal: This was the result of the bad merge, this file was untouched by me. We are working and restore.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantDataStore.hold (hunk 15 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantDataStore.hold]: This was a merge issue this file is being removed. Thanks for the catch.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Qdrant/QdrantMemoryStore.cs (hunk 14 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/Qdrant/QdrantMemoryStore.cs:34]: Updating this file with copied file.

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs (hunk 18 above)
Comment (lemillermicrosoft) [reply dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs]: Refresh should have this right.

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs (hunk 10 above)
Comment (lemillermicrosoft) [reply dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs]: Refresh should have this right.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/IVectorDb.cs [hunk 21]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+
+namespace Qdrant.DotNet;
```
[the remark this answers is not in the document]
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/IVectorDb.cs]: These files were restored and the namespaces reflected appropriately.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/IVectorDb.cs [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+
+public interface IVectorDb
```
[the remark this answers is not in the document]
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/IVectorDb.cs]: This file was restored, which shows updated namespace and internal reference of interface

File: dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs (hunk 8 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs]: This file was restored, as it was not touched in this PR.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs (hunk 20 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/QdrantMemoryStore.cs]: Yes this was a part of the initial test. Per our conversation, will retest with updates made to this PR.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Http/Specs/CreateCollectionRequest.cs (hunk 19 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.VectorDB/Qdrant/DotNet/Internal/Http/Specs/CreateCollectionRequest.cs]: Updated.

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+
+using Microsoft.SemanticKernel.AI.OpenAI.Services;
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs:3]: per previous comment - These files and other in this test project are bad merge results, I believe you just need to delete this and BackendConfigTests, OpenAIConfigTests, and CodeBlockTests.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs [hunk 24]
```diff
@@ -0,0 +1,374 @@
...
+        string collectionName,
+        Embedding<TEmbedding> target,
+        int top = 1,
+        string[]? requiredTags = null)
+    {
+        this._log.LogDebug("Searching top {0} closest vectors in {1}", top);
+
+        Verify.NotNull(target, "The given vector is NULL");
+
+        /*using HttpRequestMessage request = SearchVectorsRequest<TEmbedding>
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs:270]: please remove unused, commented code

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs [hunk 25]
```diff
@@ -0,0 +1,374 @@
...
+
+            result.Add((record, v.Score ?? 0));
+        }
+
+        // Qdrant search results are currently sorted by id, alphabetically
+        result = SortSearchResultByScore(result); */
+        //var result;
+        //foreach (var kv in result)
+        IAsyncEnumerable<(QdrantVectorRecord<TEmbedding>, double)> result = null;
+        await foreach (var kv in result)
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs:308]: There should be a null handler here

**Review by awharrison-28 — CHANGES_REQUESTED** [review]

File: dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs (hunk 23 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Test/AI/OpenAI/Services/AzureOpenAIConfigTests.cs:3]: @awharrison-28 Let me to do clean pull from main to a branch and copy all the new files fresh to ensure that there is not file overlap another bad merge.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs (hunk 24 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs:270]: @awharrison-28 I think this is from bad merge. Let me do a clean pull now and update files.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs (hunk 25 above)
Comment (tawalke) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs:308]: Will update.

**Review by shawncal — APPROVED** [review]
